mdsuite/calculators/coordination_number_calculation.py

```diff
@@ -109,267 +140,279 @@ def __init__(self, **kwargs):
 
     @call
     def __call__(
         self,
+        rdf_data: Computation = None,
         plot: bool = True,
-        data_range: int = 1,
         savgol_order: int = 2,
         savgol_window_length: int = 17,
+        number_of_shells: int = 1,
     ):
         """
 
         Parameters
         ----------
+        rdf_data : Computation (optional)
+                MDSuite Computation data schema from which to load the RDF data and
+                store relevant SQL meta-data information. If not give, an RDF will be
+                computed using the default RDF arguments.
         plot : bool (default=True)
                             Decision to plot the analysis.
-        data_range : int (default=500)
-                            Range over which the property should be evaluated.
-                            This is not applicable to the current analysis as
-                            the full rdf will be calculated.
         savgol_order : int
                 Order of the savgol polynomial filter
         savgol_window_length : int
                 Window length of the savgol filter.
-
-        Returns
-        -------
-        None.
+        number_of_shells : int
+                Number of shells to look for.
         """
-        self.plot = plot
+
+        if isinstance(rdf_data, Computation):
+            self.rdf_data = rdf_data
+        else:
+            self.rdf_data = self.experiment.run.RadialDistributionFunction(plot=False)
 
         # set args that will affect the computation result
         self.args = Args(
-            savgol_order=savgol_order, savgol_window_length=savgol_window_length
+            savgol_order=savgol_order,
+            savgol_window_length=savgol_window_length,
+            number_of_bins=self.rdf_data.computation_parameter["number_of_bins"],
+            cutoff=self.rdf_data.computation_parameter["cutoff"],
+            number_of_configurations=self.rdf_data.computation_parameter[
+                "number_of_configurations"
+            ],
+            number_of_shells=number_of_shells,
         )
 
-    def _get_density(self, species: str) -> float:
-        """
-        Use the species_tuple in front of the name to get information about the pair
+        # Auto-populate the result keys.
+        for i in range(self.args.number_of_shells):
+            self.result_keys.append(f"CN_{i + 1}")
+            self.result_keys.append(f"CN_{i + 1}_error")
+
+        self._compute_nm_volume()
+
+        self.plot = plot
+
+    def _compute_nm_volume(self):
         """
+        Compute the volume of the box in nm.
 
-        species = species.split("_")  # get an array of the species being studied
-        rdf_number_of_atoms = self.experiment.species[species[0]].n_particles
+        Returns
+        -------
+        Updates the volume attribute of the class.
+        """
+        volume_si = self.experiment.volume * (self.experiment.units["length"] ** 3)
```

Comment:
Not related to the PR but we could consider looking more into [Pint](https://pint.readthedocs.io/en/stable/) for our unit systems.

Reply:
Perhaps a good talking point at the assembly